pkg/lib/client/openshift/client.go

```diff
@@ -0,0 +1,63 @@
+package ocp
+
+import (
+	"strconv"
+
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	liberr "github.com/konveyor/forklift-controller/pkg/lib/error"
+	core "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+// Build k8s REST configuration.
+func RestCfg(p *api.Provider, secret *core.Secret) *rest.Config {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		klog.Error("failed to get config: ", err)
+		return nil
+	}
+
+	if p.IsHost() {
+		return cfg
+	}
+
+	insecure, _ := strconv.ParseBool(string(secret.Data[api.Insecure]))
+
+	// Default to insecure if parsing fails or explicitly set
```

Comment:
it's problematic to default to insecure when parsing fails - the user asks for secure connection and we do it differently...

Reply:
yes, but how did the user ask for a secure connection if we failed to parse it?